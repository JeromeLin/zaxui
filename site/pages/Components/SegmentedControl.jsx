import Markdown from '@site/components/Markdown';

export default class Page extends Markdown {
  document = () => {
    return {
      document: require('@/components/segmented-control/demo.md'),
      className: 'segmented-control-page',
    };
  }
}
