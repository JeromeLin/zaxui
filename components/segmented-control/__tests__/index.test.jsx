import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import SegmentedControl from '../index';

describe('SegmentedControl', () => {
  it('renders default correctly', () => {
    const OnChange = jest.fn();
    const wrapper = render(
      <SegmentedControl
        shape="round"
        items={['选项1', '选项2']}
        onChange={OnChange}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders selectIndex correctly', () => {
    const OnChange = jest.fn();
    const wrapper = render(
      <SegmentedControl
        shape="round"
        selectIndex={1}
        items={['选项1', '选项2']}
        onChange={OnChange}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  
});
