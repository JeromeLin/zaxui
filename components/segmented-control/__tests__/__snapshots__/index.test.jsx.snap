// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders default correctly 1`] = `
<div
  class="za-segmented-control"
>
  <div>
    <div
      class="tab round active"
    >
      <span>
        选项1
      </span>
    </div>
    <div
      class="tab round"
    >
      <span>
        选项2
      </span>
    </div>
  </div>
</div>
`;

exports[`SegmentedControl renders selectIndex correctly 1`] = `
<div
  class="za-segmented-control"
>
  <div>
    <div
      class="tab round"
    >
      <span>
        选项1
      </span>
    </div>
    <div
      class="tab round active"
    >
      <span>
        选项2
      </span>
    </div>
  </div>
</div>
`;
